src/Microsoft.Diagnostics.EventFlow.Inputs.Etw/StandardTraceEventSession.cs

```diff
@@ -0,0 +1,90 @@
+﻿// ------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All rights reserved.
+//  Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Diagnostics.Tracing;
+using Microsoft.Diagnostics.Tracing.Session;
+using Validation;
+
+namespace Microsoft.Diagnostics.EventFlow.Inputs
+{
+    internal class StandardTraceEventSession : ITraceEventSession
+    {
+        private TraceEventSession inner;
+        private bool isProcessing;
+        private IHealthReporter healthReporter;
+
+        public StandardTraceEventSession(IHealthReporter healthReporter)
+        {
+            Requires.NotNull(healthReporter, nameof(healthReporter));
+
+            this.healthReporter = healthReporter;
+            this.inner = new TraceEventSession($"EventFlow-{nameof(EtwInput)}-{Guid.NewGuid().ToString()}", TraceEventSessionOptions.Create);
+            this.inner.Source.Dynamic.All += OnEvent;
+            this.isProcessing = false;
+        }
+
+        public event Action<EventData> AllEvents;
```

Comment:
AllEvents serves a purpose that whenever the trace event happens, it got invoked. It is exposed as a public event here. At least one event handler is required, right? I feel it is a bit over exposed and it makes me feel exhausted to do null check every time is is used.
Can we make it a a private field as a simple delegate, that be passed in as a parameter in the constructor? We can then simply do null check once in the constructor and we don't need to worry about null reference in Process() or OnEvent().